components/api/idp/validator.bal

```diff
@@ -20,62 +20,75 @@ import ballerina/config;
 import ballerina/http;
 import ballerina/log;
 import ballerina/time;
-import ballerina/io;
 
 boolean isIntrospectionEPInitialized = false;
 
-http:Client introspectionEP = new("https://wso2.com" , config = {
-    auth: {
-        scheme: http:BASIC_AUTH,
-        config: {
-            username: "",
-            password: ""
-        }
-    }
-});
+public type TokenDetail record { 
+    string username;
+    int expiryTime;
+};
+
+// http:Client introspectionEP3 = new(config:getAsString(constants:IDP_INTROSPCET_VAR));
+
 # Description
 #
 # + token - access token to be validated
-# + conf - authorization configuration struct
-# + return - returns whether the access token is valid or not
-public function validateAndGetUsername(string token, string username, Conf conf) returns (string)|error {
+# + username - username to be validated
+# + return - returns a token detail if an access token is valid otherwise retuns a error
+public function validateAndGetTokenDetails(string token, string username) returns (TokenDetail)|error {
     log:printDebug("Access token validator reached and token will be validated");
+    // TODO Need to remove this after the bug on this in ballerina is fixed
     if !isIntrospectionEPInitialized {
-        var endPointUrl = conf.introspectionEp;
-        introspectionEP = new(endPointUrl , config = {
-        secureSocket: {
-            verifyHostname :false,
-            trustStore: {
-                path: config:getAsString("security.truststore"),
-                password: config:getAsString("security.truststorepass")
-            }
-        },
-        auth: {
-            scheme: http:BASIC_AUTH,
-            config: {
-                username: conf.username,
-                password: conf.password
+        var endPointUrl = io:sprintf("%s%s", config:getAsString(constants:IDP_ENDPOINT_VAR), 
+        config:getAsString(constants:IDP_INTROSPCET_VAR));
+        idpClientEP = new(endPointUrl , config = {
+            secureSocket: {
+                verifyHostname :false,
+                trustStore: {
+                    path: config:getAsString("security.truststore"),
+                    password: config:getAsString("security.truststorepass")
+                }
+            },
+            auth: {
+                scheme: http:BASIC_AUTH,
+                config: {
+                    username: config:getAsString("idp.username"),
+                    password: config:getAsString("idp.password")
+                }
             }
-        }
-    });
+        });
     isIntrospectionEPInitialized = true;
     }
+
     http:Request req = new;
     req.setPayload(io:sprintf("token=" + token));
     error ? x = req.setContentType(constants:APPLICATION_URL_ENCODED_CONTENT_TYPE);
-    var response = introspectionEP->post("", req);
+    var response = idpClientEP->post("", req);
     var isValid = false;
     if (response is http:Response) {
         json result = check response.getJsonPayload();
```

Comment:
Shall we check what the status code is ? if it's not the expected status code (2XX) let's print an error . So that we know something went wrong.

Reply:
Fixed